doc/parsing-extensions.md

```diff
@@ -0,0 +1,158 @@
+    {
+        // Directly access or modify pipeline.Extensions here, with the ability to
+        // search for other extensions, insert before or after, remove other extensions,
+        // or modify their settings.
+
+        // ...
+
+        return pipeline;
+    }
+}
+
+```
+
+### Simple Extension Example
+
+An example of a simple extension which does not add any new parsers, but instead creates a new, horrific emphasis tag, marked by triple percentage signs.  This example is based on [CitationExtension.cs](https://github.com/xoofx/markdig/blob/master/src/Markdig/Extensions/Citations/CitationExtension.cs)
+
+```csharp
+/// <summary> 
+/// An extension which applies to text of the form %%%text%%%
+/// </summary>
+public class BlinkExtension : IMarkdownExtension
+{
+    // This setup method will be run when the pipeline builder's `Build()` method is invoked. As this
+    // is a simple, self-contained extension we won't be adding anything new, but rather finding an 
+    // existing parser already in the pipeline and adding some settings to it.
+    public void Setup(MarkdownPipelineBuilder pipeline)
+    {
+        // We check the pipeline builder's inline parser collection and see if we can find a parser
+        // registered of the type EmphasisInlineParser. This is the parser which nominally handles
+        // bold and italic emphasis, but we know from its documentation that it is a general parser
+        // that can have new characters added to it.
+        var parser = pipeline.InlineParsers.FindExact<EmphasisInlineParser>();
+
+        // If we find the parser and it doesn't already have the % character registered, we add
+        // a descriptor for 3 consecutive % signs. This is specific to the EmphasisInlineParser and
+        // is just used here as an example.
+        if (parser is not null && !parser.HasEmphasisChar('%'))
+        {
+            parser.EmphasisDescriptors.Add(new EmphasisDescriptor('%', 3, 3, false));
+        }
+    }
+
+    // This method is called by the pipeline before rendering, which is a separate operation from 
+    // parsing. This implementation is just here for the purpose of the example, in which we 
+    // daisy-chain a delegate specific to the EmphasisInlineRenderer to cause an unconscionable tag 
+    // to be inserted into the HTML output wherever a %%% annotated span was placed in the source.
+    public void Setup(MarkdownPipeline pipeline, IMarkdownRenderer renderer)
+    {
+        if (renderer is not HtmlRenderer) return;
+
+        var emphasisRenderer = renderer.ObjectRenderers.FindExact<EmphasisInlineRenderer>();
+        if (emphasisRenderer is null) return;
+
+        var previousTag = emphasisRenderer.GetTag;
+        emphasisRenderer.GetTag = inline =>
+            (inline.DelimiterCount == 3 && inline.DelimiterChar == '%' ? "blink" : null)
+            ?? previousTag(inline);
+    }
+}
+```
+
+## Parsers
```

Comment:
I would probably move this part along the parsing-overview.md or another document.

In order to explain the `BlockParser`, `InnlineParser` we need to explain it top-down, from:
- `MarkdownParser.Parse` (e.g 1st pass to parse blocks by reading lines, 2nd pass to parse the content of each block for inline elements)
- `BlockProcessor.ProcessLine`
- `InlineProcessor.ProcessInlineLeaf`

Worth mentioning the https://spec.commonmark.org/0.30/#appendix-a-parsing-strategy as it has influenced the parsing of Markdig, even though in practice, Markdig had to go into more details, making it more complicated than what is described in this link.